src/wallet/rpcdump.cpp

```diff
@@ -151,9 +151,16 @@ UniValue importprivkey(const JSONRPCRequest& request)
         CKeyID vchAddress = pubkey.GetID();
         {
             pwallet->MarkDirty();
-            // We don't know which corresponding address will be used; label them all
+
+            // We don't know which corresponding address will be used;
+            // label all new addresses, and label existing addresses if a
+            // label was passed.
             for (const auto& dest : GetAllDestinationsForKey(pubkey)) {
-                pwallet->SetAddressBook(dest, strLabel, "receive");
+                if (!request.params[1].isNull()) {
+                    pwallet->SetAddressBook(dest, strLabel, "receive");
+                } else if (pwallet->mapAddressBook.count(dest) == 0) {
```

Comment:
Given that the branches for the two `if` statements are identical, why write it as a single `if (!request.params[1].isNull() || pwallet->mapAddressBook.count(dest)) {` ?

Reply:
Just to try to make it clear in the code that the `request.params[1]` has precedence. Bad idea?

Reply:
No reason besides it since AFAIK putting the look up as second condition in the `||` statement will not call it if there is a `params[1]` in the first condition.